### DISCUSSION TITLE
Visualization of DAMASK simulation results using MTEX

### DISCUSSION THREAD

**Discussion by JinkuiMeng** [mtex-toolbox/mtex#664] [Ask Anything]
From the results of damask, I can get the position of each resolution point and the corresponding Euler angle data, which is in the following form,
	X	Y	Euler1	Euler2	Euler3
0	0.5	0.5	346.99406399472707	127.49849508036522	245.7595506154353
1	1.5	0.5	346.6362510369948	127.38643418926635	245.23741240630542
2	2.5	0.5	347.17937961858485	127.68623813641577	245.8739865962412
3	3.5	0.5	346.73845509451684	127.54195881233505	245.16764070565688
4	4.5	0.5	347.34252163256167	127.80119671898086	246.03208490567826
5	5.5	0.5	347.3559226471002	127.80596228570222	246.06483770626926
6	6.5	0.5	346.9019907232955	127.66930016775669	245.3213939916073
7	7.5	0.5	346.8021487755355	127.69992356766348	245.15997907883133
8	8.5	0.5	346.99834420677115	127.72625282724906	245.47364001530232
....

Then, I modified the file (testdata_hex.ctf) given by MTeX software, that is, I replaced the above X, Y, Euler 1, Euler 2 and Euler 3 respectively, and supplemented the rest with 0. Part of the result is as follows,

Channel Text File												
Prj	testdata.ctf											
Author	[Unknown]											
JobMode	Grid											
XCells	30											
YCells	30											
XStep	4.25											
YStep	4.25											
AcqE1	0											
AcqE2	0											
AcqE3	0											
Euler angles refer to Sample Coordinate system (CS0)!	Mag	64	Coverage	100	Device	0	KV	17	TiltAngle	70	TiltAxis	0
Phases	1											
9.541;17.74;5.295	90;103.67;90	Ti	2	0	0_5.0.9.1	1477773286	[Ti.cry]					
Phase	X	Y	Bands	Error	Euler1	Euler2	Euler3	MAD	BC	BS		
1	0.5	0.5	7	0	346.994064	127.4984951	245.7595506	0	0			
1	1.5	0.5	7	0	346.636251	127.3864342	245.2374124	0	0			
1	2.5	0.5	7	0	347.1793796	127.6862381	245.8739866	0	0			
1	3.5	0.5	7	0	346.7384551	127.5419588	245.1676407	0	0			
1	4.5	0.5	7	0	347.3425216	127.8011967	246.0320849	0	0			
1	5.5	0.5	7	0	347.3559226	127.8059623	246.0648377	0	0			
1	6.5	0.5	7	0	346.9019907	127.6693002	245.321394	0	0			
1	7.5	0.5	7	0	346.8021488	127.6999236	245.1599791	0	0			
1	8.5	0.5	7	0	346.9983442	127.7262528	245.47364	0	0			
1	9.5	0.5	7	0	346.6747686	127.6791058	244.8968484	0	0			
1	10.5	0.5	7	0	346.4779736	127.6481018	244.6477495	0	0		
...

Then, i use the code as follows,

fileName = [mtexDataPath filesep 'EBSD' filesep 'Titanium_hex.ctf'];
ebsd = EBSD.load(fileName)

plotx2east % this command tell MTEX to plot the x coordinate pointing towards east
figure(1)
plot(ebsd,'coordinates','on')

ebsd('Ti')
ebsd('Ti').orientations

figure(2)
plot(ebsd('Ti'),ebsd('Ti').orientations,'micronbar','off')

grains = calcGrains(ebsd('indexed'),'theshold',10*degree)
grains = smooth(grains,4);
figure(2)
hold on
plot(grains.boundary,'lineWidth',2)
hold off

% define the crystal shape of Ti and store it in the variable cS
**cS = crystalShape.hex(ebsd('Ti').CS)**

because I want to plot at the positions of the Ti grains the crystal shapes(https://mtex-toolbox.github.io/images/EBSDTutorial_05.png).

BUT this bold statement is problematic. 

Is it because these three parameters(MAD/BC/BS) are all defined as 0?

![bug](https://user-images.githubusercontent.com/64880536/103441120-b622e180-4c86-11eb-98bc-27c5337a2972.jpg)
![IPF with GB](https://user-images.githubusercontent.com/64880536/103441122-b9b66880-4c86-11eb-9038-5fc58acdaca7.jpg)
![IPF](https://user-images.githubusercontent.com/64880536/103441123-bae79580-4c86-11eb-8c82-b28b069f52a2.jpg)

**Comment by kilir** [comment]
Hi,
you can load your data using the import wizard (within which you can specify your crystal symmetry as well). No need to fudge it into a ctf. What is the output of `ebsd('Ti').CS`. I suspect you have a crystalSymmetry of `1` instead of `622` or somethign that would work on hexagonal symmetry.

Cheers,
Rüdiger
